develop/appdir/app/styles-provider.tsx

```diff
@@ -0,0 +1,21 @@
+'use client'
+import { Main } from 'next/document'
+import { useServerInsertedHTML } from 'next/navigation'
+import { useState } from 'react'
+import { AppRegistry } from 'react-native'
+
+export function StylesProvider({ children }: { children: React.ReactNode }) {
+    const [getStyleElement] = useState(() => {
+      AppRegistry.registerComponent('Main', () => Main)
+      // @ts-ignore
+      const { getStyleElement } = AppRegistry.getApplication('Main')
+  
+      return getStyleElement
+    })
+    useServerInsertedHTML(() => {
+      return getStyleElement()
+    })
```

Comment:
Looks like using `AppRegistry` is entirely pointless. I just got this to work instead.

```suggestion
import { StyleSheet } from 'react-native';

export function StylesProvider({ children }: { children: React.ReactNode }) {
  useServerInsertedHTML(() => {
    // @ts-ignore
    const sheet = StyleSheet.getSheet();
    return (
      <style
        dangerouslySetInnerHTML={{ __html: sheet.textContent }}
        id={sheet.id}
      />
    );
  });
```

Reply:
I nicked this code from https://github.com/necolas/react-native-web/blob/c43b629cba262d13a44b88eeb44ac72fbb5b28af/packages/react-native-web/src/exports/AppRegistry/renderApplication.js#LL68C1-L68C1

Reply:
oh, wow, cool. will try this out

Reply:
Updated